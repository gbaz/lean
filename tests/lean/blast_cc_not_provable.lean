set_option blast.strategy "cc"

example (C : nat → Type) (f : Π n, C n → C n) (n m : nat) (c : C n) (d : C m) :
        f n == f m → c == d → f n c == f m d :=
by blast
