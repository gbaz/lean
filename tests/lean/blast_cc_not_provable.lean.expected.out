blast_cc_not_provable.lean:5:0: error: blast tactic failed
strategy 'cc' failed, no proof found, final state:
C : ℕ → Type,
n m : ℕ,
f : Π (n : ℕ), C n → C n,
c : C n,
d : C m,
H.6 : f n == f m,
H.7 : c == d
⊢ f n c == f m d
-------
proof state:
C : ℕ → Type,
f : Π (n : ℕ), C n → C n,
n m : ℕ,
c : C n,
d : C m
⊢ f n == f m → c == d → f n c == f m d
blast_cc_not_provable.lean:5:0: error: don't know how to synthesize placeholder
C : ℕ → Type,
f : Π (n : ℕ), C n → C n,
n m : ℕ,
c : C n,
d : C m
⊢ f n == f m → c == d → f n c == f m d
blast_cc_not_provable.lean:5:0: error: failed to add declaration 'example' to environment, value has metavariables
remark: set 'formatter.hide_full_terms' to false to see the complete term
  ?M_1
